### PR TITLE
Manage sig indexes

### PR DESCRIPTION
**Pull request by l3wi**

### CODE REVIEW

File: lib/transfer.js [hunk 1]
```diff
@@ -306,7 +307,7 @@ function getDiff(root, remainder, history, bundles) {
   if(!history) {
     throw new Error(TransferErrors.NULL_VALUE);
   }
-  if(!bundlesd) {
+  if(!bundles) {
```
Comment (paulhandy) [inline lib/transfer.js:309]: 👍

File: lib/transfer.js [hunk 2]
```diff
@@ -364,6 +365,7 @@ module.exports = {
   'close'          : close,
   'getDiff'        : getDiff,
   'sign'           : sign,
+  'appliedSignatures': appliedSignatures,
```
Comment (paulhandy) [inline lib/transfer.js:367]: 💯

File: lib/transfer.js [hunk 3]
```diff
@@ -244,14 +245,14 @@ function sign(root, seed, bundles) {
...
       address: multisig.address,
-      index: multisig.signingIndex,
+      index: signingIndex,
       signatureFragments: IOTACrypto.multisig.addSignature(
         bundle, 
         multisig.address, 
         IOTACrypto.multisig.getKey(seed, multisig.index, multisig.security))
       .filter(tx => tx.address == multisig.address)
       .map(tx => tx.signatureMessageFragment)
-      .slice(multisig.signingIndex, multisig.security)
```
Comment (paulhandy) [inline lib/transfer.js]: keep this but fix it

File: lib/transfer.js [hunk 4]
```diff
@@ -225,13 +225,14 @@ function getMultisigs(root, transfers) {
  *
  * @return {[{object}]} signatures
  */
-function sign(root, seed, bundles) {
```
Comment (paulhandy) [inline lib/transfer.js]: keep this one

File: lib/transfer.js [hunk 5]
```diff
@@ -225,13 +225,14 @@ function getMultisigs(root, transfers) {
...
+function sign(root, seed, bundles, userIndex) {
   const multisigs = getMultisigs(root, bundles);
   return helpers.deepClone(bundles).map((bundle, i) => {
     const multisig = multisigs[i];
     // multisig has member signingIndex
+    let signingIndex = userIndex * multisig.security
```
Comment (paulhandy) [inline lib/transfer.js]: remove this

File: lib/transfer.js [hunk 6]
```diff
@@ -225,13 +225,14 @@ function getMultisigs(root, transfers) {
...
  */
-function sign(root, seed, bundles) {
+function sign(root, seed, bundles, userIndex) {
   const multisigs = getMultisigs(root, bundles);
   return helpers.deepClone(bundles).map((bundle, i) => {
     const multisig = multisigs[i];
     // multisig has member signingIndex
+    let signingIndex = userIndex * multisig.security    
     let signedTx = bundle.filter(tx => tx.address == multisig.address);
-    for(let j = 0; j < multisig.signingIndex; j++) {
```
Comment (paulhandy) [inline lib/transfer.js]: keep this

File: lib/transfer.js [hunk 7]
```diff
@@ -244,14 +245,14 @@ function sign(root, seed, bundles) {
     }
     return { bundle: bundle[0].bundle,
       address: multisig.address,
-      index: multisig.signingIndex,
```
Comment (paulhandy) [inline lib/transfer.js]: this

File: lib/transfer.js [hunk 8]
```diff
@@ -266,7 +267,7 @@ function appliedSignatures(bundles, signatures) {
       let addy = bundle.find(tx => tx.value < 0 ).address;
       bundle
         .filter(tx => tx.address == addy)
-        .slice(userSignature.index, userSignature.signatureFragments.length)
+        .slice(userSignature.index, userSignature.index + userSignature.signatureFragments.length)
```
Comment (paulhandy) [inline lib/transfer.js:269]: 👍

**Review by paulhandy — CHANGES_REQUESTED** [review]

File: lib/transfer.js (hunk 8 above)
Comment (l3wi) [reply lib/transfer.js:269]: 🥔

File: lib/transfer.js (hunk 1 above)
Comment (l3wi) [reply lib/transfer.js:309]: 🥔

File: lib/transfer.js (hunk 2 above)
Comment (l3wi) [reply lib/transfer.js:367]: 🚰

**Review by paulhandy — APPROVED** [review]

**Pull request merged**